editor/editor_config.cpp

```diff
@@ -120,9 +120,22 @@ EditorConfig::EditorConfig(string const & fileName)
   Reload();
 }
 
-bool EditorConfig::GetTypeDescription(vector<string> const & classificatorTypes,
+bool EditorConfig::GetTypeDescription(vector<string> && classificatorTypes,
                                       TypeAggregatedDescription & outDesc) const
 {
+  bool isBuilding = false;
+  for (auto it = classificatorTypes.begin(); it != classificatorTypes.end(); ++it)
+  {
+    if (*it == "building")
+    {
+      outDesc.m_address = isBuilding = true;
+      outDesc.m_editableFields.push_back(feature::Metadata::FMD_BUILDING_LEVELS);
+      outDesc.m_editableFields.push_back(feature::Metadata::FMD_POSTCODE);
+      classificatorTypes.erase(it);
```

Comment:
Эта строчка убирает заматчившийся building из вектора всех типов фичи, чтобы по нему зря не бегать дальше в циклах.

> On Apr 1, 2016, at 16:02, Ilya Zverev <notifications@github.com> wrote:
> 
> In editor/editor_config.cpp <https://github.com/mapsme/omim/pull/2670#discussion_r58201922>:
> 
> >                                        TypeAggregatedDescription & outDesc) const
> >  {
> > +  bool isBuilding = false;
> > +  for (auto it = classificatorTypes.begin(); it != classificatorTypes.end(); ++it)
> > +  {
> > +    if (*it == "building")
> > +    {
> > +      outDesc.m_address = isBuilding = true;
> > +      outDesc.m_editableFields.push_back(feature::Metadata::FMD_BUILDING_LEVELS);
> > +      outDesc.m_editableFields.push_back(feature::Metadata::FMD_POSTCODE);
> > +      classificatorTypes.erase(it);
> Вот эта строчка зачем? Она не сломает какой-то важный список в приложении?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2670/files/af10e69d522742081c4007f6da63a022705f438b#r58201922>